### PR TITLE
WebKit no longer needs to take unbounded network assertions on the network process for downloads

### PR DESCRIPTION
**Pull request by cdumez**
#### b95877fdcedb339c3460243d143bd20dc0118e59
<pre>
WebKit no longer needs to take unbounded network assertions on the network process for downloads
<a href="https://bugs.webkit.org/show_bug.cgi?id=280828">https://bugs.webkit.org/show_bug.cgi?id=280828</a>
<a href="https://rdar.apple.com/133715239">rdar://133715239</a>

Reviewed by Per Arne Vollan.

WebKit no longer needs to take unbounded network assertions on the network process for downloads
now that dasd is taking this assertion for us.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/NetworkProcess/Downloads/DownloadMap.cpp: Removed.
* Source/WebKit/NetworkProcess/Downloads/DownloadMap.h:
* Source/WebKit/NetworkProcess/NetworkResourceLoadMap.h:
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp:
(WebKit::DownloadProxyMap::createDownloadProxy):
(WebKit::DownloadProxyMap::downloadFinished):
(WebKit::DownloadProxyMap::invalidate):
* Source/WebKit/UIProcess/Downloads/DownloadProxyMap.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/284764@main">https://commits.webkit.org/284764@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/61111922490320804cbacdece0bcc2661a171abc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/70031 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/49432 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22791 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/74116 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/21197 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/57238 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/21040 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55576 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/14055 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/73097 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/45038 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/60410 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/36042 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41698 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/17841 "Found 1 new test failure: fullscreen/full-screen-document-background-color.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/19566 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/63150 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63622 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/18188 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75834 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/69277 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/14256 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/17426 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/63277 "Found 1 new test failure: webanimations/accelerated-translate-animation-with-transform.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/14298 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/60479 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63200 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/11225 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4845 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/91059 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10787 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/45235 "Built successfully") | [⏳ 🛠 mac-safer-cpp ](https://ews-build.webkit.org/#/builders/macOS-Safer-CPP-Checks-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/19859 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/46309 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/47588 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/46050 "Built successfully") | | |

### CODE REVIEW

**Review by pvollan — APPROVED** [review]
Nice! R=me.

File: Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp [hunk 1]
```diff
@@ -86,10 +86,8 @@ Ref<DownloadProxy> DownloadProxyMap::createDownloadProxy(WebsiteDataStore& dataS
 
     if (m_downloads.size() == 1 && m_shouldTakeAssertion) {
         ASSERT(!m_downloadUIAssertion);
-        ASSERT(!m_downloadNetworkingAssertion);
         m_downloadUIAssertion = ProcessAssertion::create(getCurrentProcessID(), "WebKit downloads"_s, ProcessAssertionType::UnboundedNetworking);
-        m_downloadNetworkingAssertion = ProcessAssertion::create(protectedProcess().get(), "WebKit downloads"_s, ProcessAssertionType::UnboundedNetworking);
```
Comment (pvollan) [inline Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp:91]: Do we still need this assertion on macOS?

Comment (cdumez) [reply Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp:91]: I *thought* we never suspended the network process on macOS but we should check.

Comment (pvollan) [reply Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp:91]: Ah, yes, you are right, we take a background activity in NetworkProcessProxy:

```
#if PLATFORM(MAC)
    , m_backgroundActivityToPreventSuspension(throttler().backgroundActivity("Prevent suspension"_s))
#endif
```

**Pull request merged**